dev/archery/archery/linking.py

```diff
@@ -61,9 +63,84 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _extract_symbols(self, symbol_info):
+        return [re.search(r'\S+$', line).group() for line in symbol_info if line]
+
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not re.search(r'\s[Ww]\s', line)]
+
+    def _capture_symbols(self, remove_symbol_versions, symbol_info):
+        if remove_symbol_versions:
+            symbol_info = [re.split('@@', line)[0] for line in symbol_info]
+        return symbol_info
+
+    def list_symbols_for_dependency(self, dependency, remove_symbol_versions=False):
+        if dependency == 'linux-vdso.so.1':
+            return []
```

Comment:
Why? Can you add a comment?

Reply:
Explained here: https://github.com/apache/arrow/pull/40520#discussion_r1555272123
And I added a comment too.